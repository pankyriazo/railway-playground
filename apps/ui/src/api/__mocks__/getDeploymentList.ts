import { GetDeploymentListResponse } from "@/api/types";

export const mockGetDeploymentList = {
  data: {
    deployments: {
      edges: [
        {
          node: {
            id: "4b049de0-1ce0-4e81-8036-26b252e3285c",
            status: "SUCCESS",
            updatedAt: "2024-03-17T08:50:14.973Z",
            environmentId: "632da482-c5b1-4570-87af-9ea225c335cb",
          },
        },
        {
          node: {
            id: "38b68be4-f86d-4987-8c48-dee293fda404",
            status: "SUCCESS",
            updatedAt: "2024-03-16T19:41:25.747Z",
            environmentId: "1beb4373-6b1f-43cb-8150-0e7214acea82",
          },
        },
      ],
    },
  },
} as const satisfies GetDeploymentListResponse;

export const mockGetDeploymentListError = {
  data: null,
  errors: [
    {
      message: "Unauthorized",
    },
  ],
} as const satisfies GetDeploymentListResponse;
