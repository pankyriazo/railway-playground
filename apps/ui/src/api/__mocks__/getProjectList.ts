import { GetProjectListResponse } from "@/api/types";

export const mockGetProjectList = {
  data: {
    me: {
      projects: {
        edges: [
          {
            node: {
              id: "a20a0d44-a6de-4804-acf1-58d436570fd7",
              name: "nauseating-cup",
              services: {
                edges: [
                  {
                    node: {
                      id: "0e62e7bf-2008-4452-bd32-e57cec3bb9a2",
                      name: "fascinated-motion",
                    },
                  },
                  {
                    node: {
                      id: "7ab42902-aae8-47b7-9976-81496dc61834",
                      name: "Postgres",
                    },
                  },
                ],
              },
              plugins: {
                edges: [],
              },
              environments: {
                edges: [
                  {
                    node: {
                      id: "1beb4373-6b1f-43cb-8150-0e7214acea82",
                      name: "production",
                    },
                  },
                  {
                    node: {
                      id: "632da482-c5b1-4570-87af-9ea225c335cb",
                      name: "staging",
                    },
                  },
                ],
              },
            },
          },
        ],
      },
    },
  },
} as const satisfies GetProjectListResponse;

export const mockGetProjectListError = {
  data: null,
  errors: [
    {
      message: "Unauthorized",
    },
  ],
} as const satisfies GetProjectListResponse;
