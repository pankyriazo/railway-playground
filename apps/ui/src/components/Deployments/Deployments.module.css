.title {
  color: var(--color-text--600);
  font-size: var(--font-size-md);
  font-weight: var(--font-weight-medium);
  margin-bottom: var(--spacing-sm);
}

.list {
  display: flex;
  flex-direction: column;
  row-gap: var(--spacing-xs);
}

.empty {
  color: var(--color-text--600);
  font-size: var(--font-size-sm);
}
