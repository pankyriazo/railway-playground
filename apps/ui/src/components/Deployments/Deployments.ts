import { Deployments as DeploymentsType } from "@monorepo/types";
import { Component } from "@/types";
import Deployment from "@/components/Deployment";

import styles from "./Deployments.module.css";

class Deployments extends Component<HTMLUListElement> {
  private _deployments: NonNullable<DeploymentsType["data"]>["deployments"];

  constructor({
    deployments,
  }: {
    deployments: NonNullable<DeploymentsType["data"]>["deployments"];
  }) {
    super();

    this._element = document.createElement("ul");
    this._element.id = "deployments";
    this._deployments = deployments;

    this.render();
  }

  override render() {
    if (!this._element) {
      return;
    }

    this._element.innerHTML = "";
    this._element.dataset.testid = "deployments";
    this._element.className = styles.list!;

    if (this._deployments.edges.length === 0) {
      const empty = document.createElement("p");
      empty.dataset.testid = "deployments__empty";
      empty.className = styles.empty!;
      empty.textContent = "No deployments found.";
      this._element.appendChild(empty);
      return;
    }

    for (const { node: deploymentData } of this._deployments.edges) {
      const deployment = new Deployment({
        deployment: deploymentData,
      });
      this._element.appendChild(deployment.element);
    }
  }
}

export default Deployments;
