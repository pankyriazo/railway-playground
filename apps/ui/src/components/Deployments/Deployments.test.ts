import { mockComponent, render } from "@/test-helpers";
import { mockGetDeploymentList } from "@/api/__mocks__/getDeploymentList";
import Deployments from "./Deployments";

jest.mock("@/components/Deployment", () => mockComponent("Deployment"));

describe("Deployments", () => {
  describe("when the deployments are empty", () => {
    it("should render empty message", () => {
      render(Deployments, {
        deployments: {
          edges: [],
        },
      });

      const empty = document.querySelector("[data-testid=deployments__empty]");

      expect(empty).not.toBeNull();
      expect(empty!.textContent).toBe("No deployments found.");
      expect(empty).toMatchSnapshot();
    });
  });

  describe("when the deployments are not empty", () => {
    it("should render deployments", () => {
      render(Deployments, {
        deployments: mockGetDeploymentList.data.deployments,
      });

      const deployments = document.querySelectorAll(
        "[data-component=Deployment]"
      );

      expect(deployments).toHaveLength(1);
      expect(deployments).toMatchSnapshot();
    });
  });
});
