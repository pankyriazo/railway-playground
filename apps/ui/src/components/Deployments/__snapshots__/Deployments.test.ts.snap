// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deployments when the deployments are empty should render empty message 1`] = `
<p
  class="empty"
  data-testid="deployments__empty"
>
  No deployments found.
</p>
`;

exports[`Deployments when the deployments are not empty should render deployments 1`] = `
NodeList [
  <div
    data-component="Deployment"
  />,
]
`;
